gdk4/src/auto/key_event.rs

```diff
@@ -0,0 +1,59 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files.git)
+// DO NOT EDIT
+
+use crate::Event;
+use crate::ModifierType;
+use glib::translate::*;
+use glib::StaticType;
+
+glib::wrapper! {
+    #[doc(alias = "GdkKeyEvent")]
+    pub struct KeyEvent(Shared<ffi::GdkKeyEvent>);
+
+    match fn {
+        ref => |ptr| ffi::gdk_event_ref(ptr as *mut ffi::GdkEvent),
+        unref => |ptr| ffi::gdk_event_unref(ptr as *mut ffi::GdkEvent),
+    }
+}
+
+impl glib::StaticType for KeyEvent {
+    fn static_type() -> glib::Type {
+        unsafe { from_glib(ffi::gdk_key_event_get_type()) }
+    }
+}
+
+impl KeyEvent {
+    #[doc(alias = "gdk_key_event_get_consumed_modifiers")]
+    #[doc(alias = "get_consumed_modifiers")]
+    pub fn consumed_modifiers(&self) -> ModifierType {
+        unsafe {
+            from_glib(ffi::gdk_key_event_get_consumed_modifiers(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gdk_key_event_get_keycode")]
+    #[doc(alias = "get_keycode")]
+    pub fn keycode(&self) -> u32 {
```

Comment:
Shouldn't this return a `gdk::Key`?

Reply:
For keyval yes, not sure about keycode

Reply:
Ah right, there's key(val) and keycode. I got confused :)